backend/src/main/java/com/htecgroup/skynest/filter/CustomAuthenticationFilter.java

```diff
@@ -0,0 +1,73 @@
+package com.htecgroup.skynest.filter;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.htecgroup.skynest.exception.UserException;
+import com.htecgroup.skynest.exception.UserExceptionType;
+import com.htecgroup.skynest.model.request.UserLoginRequest;
+import com.htecgroup.skynest.security.SecurityConstants;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+
+@AllArgsConstructor
+@Log4j2
+public class CustomAuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+
+    private final AuthenticationManager authenticationManager;
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder passwordEncoder;
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest request, HttpServletResponse response) throws AuthenticationException {
+        UserLoginRequest credentials = new UserLoginRequest();
+        try {
+            credentials = new ObjectMapper().readValue(request.getInputStream(), UserLoginRequest.class);
```

Comment:
Since ObjectMapper probably will be used in different places, you can extract it as spring Bean and inject it here

Reply:
Also, does ObjectMapper do the same thing as ModelMapper? If so we should probably use the same on everywhere.

Reply:
it doesn't work with ModelMapper, not sure why, so I've made ObjectMapper as a bean and inject it in the code.

Reply:
ObjectMapper is used when you convert json to object and vice versa, ModelMapper is used when you map from one object into another